metric/test/exporter_test.go

```diff
@@ -50,6 +52,35 @@ func ExampleExporter() {
 	// increased by 6
 }
 
+type derivedMetric struct {
+	i int64
+}
+
+func (m *derivedMetric) ToInt64() int64 {
+	return m.i
+}
+
+func ExampleExporter_metric() {
+	metricReader := metricexport.NewReader()
+	metrics := NewExporter(metricReader)
+	r := metric.NewRegistry()
+	g, _ := r.AddInt64DerivedCumulative("derived", metric.WithLabelKeys(myTag.Name()))
+	for i := 1; i <= 3; i++ {
+		// The code under test begins here.
+		m := derivedMetric{int64(i)}
+		g.UpsertEntry(m.ToInt64, metricdata.NewLabelValue("l1"))
```

Comment:
proper example of derived metrics would be move line 70 and 71 outside of the for loop and then updating `m.i = i` in the for loop.

Reply:
Thanks for the tip, I like that! I suspected that I wasn't using the API as intended. I'm glad to have this as a reference of how this is intended to be used :)